src/core/space_interaction_API.jl

```diff
@@ -371,3 +372,41 @@ function random_nearby_agent(a, model, r = 1; kwargs...)
     isnothing(id) && return
     return model[id]
 end
+
+"""
+    random_nearby_position(position, model::ABM, r=1; kwargs...) → position
+
+Return a random position near the given `position`. Return `nothing` if the space doesn't allow for nearby positions.
+
+The function uses the same Reservoir Sampling algorithm as [`random_nearby_id`](@ref). 
+
+The value of the argument `r` and possible keywords operate identically to [`nearby_positions`](@ref).
+
+"""
```

Comment:
```suggestion
"""
    random_nearby_position(position, model::ABM, r=1; kwargs...) → position

Return a random position near the given `position`. Return `nothing` if the space doesn't allow for nearby positions.

The value of the argument `r` and possible keywords operate identically to [`nearby_positions`](@ref).
"""
```

I wouldn't mentioned the Reservoir Sampling in the docstring, I think it's not relevant to the users and it isn't done in another similar docstrings, as a comment inside the function instead it seems good to me as you already did.